test/suite/test_hs21.py

```diff
@@ -0,0 +1,181 @@
+            if not c.isdigit():
+                self.assertEqual(c, '=')
+                break
+            write_gen_str = c + write_gen_str
+        return int(write_gen_str)
+
+    def test_hs(self):
+        active_files = []
+        value1 = 'a' * 500
+        value2 = 'd' * 500
+
+        # Set up 'numfiles' with 'numrows' entries. We want to create a number of files that
+        # contain active history (content newer than the oldest timestamp).
+        for f in range(self.numfiles):
+            table_uri = 'table:%s.%d' % (self.file_name, f)
+            file_uri = 'file:%s.%d.wt' % (self.file_name, f)
+            # Create a small table.
+            ds = SimpleDataSet(
+                self, table_uri, 0, key_format='S', value_format='S', config='log=(enabled=false)')
+            ds.populate()
+            # Checkpoint to ensure we write the files metadata checkpoint value.
+            self.session.checkpoint()
+            # Get the base write gen of the file so we can compare after the handles get closed.
+            base_write_gen = self.parse_run_write_gen(file_uri)
+            active_files.append((base_write_gen, ds))
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        # Perform a series of updates over our files to check the history store is
+        # working with old and new timestamps.
+        for (_, ds) in active_files:
+            # Load data.
+            self.large_updates(ds.uri, value1, ds, self.nrows // 2 , 1)
+            # Check that all updates are seen.
+            self.check(value1, ds.uri, self.nrows // 2, 1)
+
+            # Load more data with a later timestamp.
+            self.large_updates(ds.uri, value2, ds, self.nrows, 100)
+            # Check that the new updates are only seen after the update timestamp.
+            self.check(value1, ds.uri, self.nrows // 2, 1)
+            self.check(value2, ds.uri, self.nrows, 100)
+
+        # Our sweep scan interval is every 1 second and the amount of idle time needed for a handle to be closed is 3 seconds.
+        # It should take roughly 4 seconds for the sweep server to close our file handles. Lets wait at least double
+        # that to be safe.
+        max = 8
+        sleep = 0
+        # After waiting for the sweep server to remove our idle handles, the only open
+        # handles that should be the metadata file, history store file and lock file.
+        final_numfiles = 3
+        while sleep < max:
+            # We continue doing checkpoints to ensure we sweep the session cache, allowing
+            # idle handles to be removed.
+            self.session.checkpoint()
+            sleep += 0.5
+            time.sleep(0.5)
+            # Open the stats cursor to get the current dhandle sweep status.
+            stat_cursor = self.session.open_cursor('statistics:', None, None)
```

Comment:
You do not need to open/close stat cursor in a loop or that frequently. You can keep one open at the start and close at the end. If i am remembering correctly "cursor.reset()" will fetch new stats into the underlying data structures.

Reply:
Good point, wasn't aware `cursor.reset()` fetched new stats. I've updated it to avoid the continuous open/close of the stat cursor.